src/nvim/log.c

```diff
@@ -116,6 +121,14 @@ void log_unlock(void)
   uv_mutex_unlock(&mutex);
 }
 
+static void on_log_recursive_event(void **argv)
+{
+  char *fn_name = argv[0];
+  ptrdiff_t linenr = (ptrdiff_t)argv[1];
+  siemsg("E5430: %s:%d: recursive log!", fn_name, linenr);
```

Comment:
show a user error (once) instead of logging to stderr (which is unpleasant for embedders?)